alignak/commandcall.py

```diff
@@ -83,39 +76,53 @@ class CommandCall(DummyCommandCall):
         'enable_environment_macros': BoolProp(default=False),
     }
 
-    def __init__(self, commands, call, poller_tag='None',
-                 reactionner_tag='None', enable_environment_macros=False):
-        self._id = self.__class__._id
-        self.__class__._id += 1
-        self.call = call
-        self.timeout = -1
-        # Now split by ! and get command and args
-        self.get_command_and_args()
-        self.command = commands.find_by_name(self.command.strip())
-        self.late_relink_done = False  # To do not relink again and again the same commandcall
-        self.valid = self.command is not None
-        self.enable_environment_macros = enable_environment_macros
-        if self.valid:
-            # If the host/service do not give an override poller_tag, take
-            # the one of the command
-            self.poller_tag = poller_tag  # from host/service
-            self.reactionner_tag = reactionner_tag
-            self.module_type = self.command.module_type
-            self.enable_environment_macros = self.command.enable_environment_macros
-            self.timeout = int(self.command.timeout)
-            if self.valid and poller_tag is 'None':
-                # from command if not set
-                self.poller_tag = self.command.poller_tag
-            # Same for reactionner tag
-            if self.valid and reactionner_tag is 'None':
-                # from command if not set
-                self.reactionner_tag = self.command.reactionner_tag
+    def __init__(self, params):
+
+        if 'commands' in params:
+            commands = params['commands']
+            self.call = params['call']
+            self.enable_environment_macros = params.get('enable_environment_macros', False)
+            self.uuid = uuidmod.uuid4().hex
+            self.timeout = -1
+            command, self.args = self.get_command_and_args()
+            self.command = commands.find_by_name(command)
+            self.late_relink_done = False  # To do not relink again and again the same commandcall
+            self.valid = self.command is not None
+            if self.valid:
+                # If the host/service do not give an override poller_tag, take
+                # the one of the command
+                self.poller_tag = params.get('poller_tag', 'None')  # from host/service
+                self.reactionner_tag = params.get('reactionner_tag', 'None')
+                self.module_type = self.command.module_type
+                self.enable_environment_macros = self.command.enable_environment_macros
+                self.timeout = int(self.command.timeout)
+                if self.valid and params.get('poller_tag', 'None') == 'None':
+                    # from command if not set
+                    self.poller_tag = self.command.poller_tag
+                # Same for reactionner tag
+                if self.valid and params.get('reactionner_tag', 'None') == 'None':
```

Comment:
Why not use directly _self.reactionner_tag_ instead use: 
`params.get('reactionner_tag', 'None')`. It will be more simple ;)